### PR TITLE
Implement pre-undeploy-event-reg hook to undeploy event registrations

### PR DESCRIPTION
**Pull request by sangeetha5491**
## Description

Undeploy event registrations in the pre-undeploy-event-reg hook

## Related Issue

- https://github.com/adobe/aio-cli-plugin-events/pull/64
- https://github.com/adobe/aio-cli-plugin-app/pull/694

## Is Blocked By

- release of https://github.com/adobe/aio-cli-plugin-app/pull/694

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: src/hooks/utils/hook-utils.js [hunk 1]
```diff
@@ -224,9 +224,50 @@ async function deployRegistration ({ appConfig: { events, project } }, expectedD
   }
 }
 
+/**
+ * @param {object} appConfigRoot - Root object containing events and project details
+ * @param {object} appConfigRoot.appConfig - Object containing events and project details
+ * @param {object} appConfigRoot.appConfig.project - Project details from the .aio file
+ * @param {object} appConfigRoot.appConfig.events - Events registrations that are part of the app.config.yaml file
+ * @param {string} hookType - pre-deploy-event-reg or post-deploy-event-reg hook values
```
Comment (purplecabbage) [inline src/hooks/utils/hook-utils.js:232]: I don't think you need this anymore ...

**Review by purplecabbage — APPROVED** [review]
lgtm!

File: src/hooks/utils/hook-utils.js (hunk 1 above)
Comment (purplecabbage) [reply src/hooks/utils/hook-utils.js:232]: `hookType` that is, since its always gonna be `pre-deploy-event-reg` now ...

Comment (sangeetha5491) [reply src/hooks/utils/hook-utils.js]: removed

**Pull request merged**